src/software/networking/proto_udp_client.h

```diff
@@ -66,6 +69,10 @@ class ProtoUdpClient
 
     static constexpr unsigned int MAX_BUFFER_LENGTH = 9000;
     std::array<char, MAX_BUFFER_LENGTH> raw_received_data_;
+
+    boost::asio::io_service::work work;
```

Comment:
This is needed because `io_service.run()` terminates immediately otherwise
https://stackoverflow.com/questions/44934987/boostasio-making-sure-io-service-run-is-invoked-only-after-async-receive

Which makes me think the `io_service.run` call in the other threaded listener/sender are not required